processing/src/main/java/io/druid/query/groupby/epinephelinae/GroupByRowProcessor.java

```diff
@@ -98,7 +112,18 @@
                                        ? BooleanValueMatcher.of(true)
                                        : filter.makeMatcher(columnSelectorFactory);
 
-    final FilteredSequence<Row> filteredSequence = new FilteredSequence<>(
+    final FilteredSequence<Row> filteredSequence = wasQueryPushedDown ? new FilteredSequence<>(
+        rows,
+        new Predicate<Row>()
+        {
+          @Override
+          public boolean apply(Row input)
+          {
+            rowSupplier.set(input);
+            return true; // nothing to filter since it has already been done on the push down
```

Comment:
If this is the case, I believe it's much faster to call `final AggregateResult retVal = rows.accumulate(AggregateResult.ok(), accumulator);` at https://github.com/apache/incubator-druid/pull/5471/files#diff-dfac707f8edd4ee002b9f3372cfc2699R192.